crates/router/src/connector/threedsecureio/transformers.rs

```diff
@@ -0,0 +1,676 @@
+        })
+    }
+}
+
+pub struct ThreedsecureioAuthType {
+    pub(super) api_key: Secret<String>,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for ThreedsecureioAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
+                api_key: api_key.to_owned(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        }
+    }
+}
+
+fn get_card_details(
+    payment_method_data: api_models::payments::PaymentMethodData,
+) -> Result<api_models::payments::Card, errors::ConnectorError> {
+    match payment_method_data {
+        api_models::payments::PaymentMethodData::Card(details) => Ok(details),
+        _ => Err(errors::ConnectorError::NotSupported {
+            message: SELECTED_PAYMENT_METHOD.to_string(),
+            connector: "threedsecureio",
+        })?,
+    }
+}
+
+impl TryFrom<&ThreedsecureioRouterData<&types::authentication::ConnectorAuthenticationRouterData>>
+    for ThreedsecureioAuthenticationRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &ThreedsecureioRouterData<&types::authentication::ConnectorAuthenticationRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let request = &item.router_data.request;
+        //browser_details are mandatory for Browser flows
+        let browser_details = match request.browser_details.clone() {
+            Some(details) => Ok::<Option<types::BrowserInformation>, Self::Error>(Some(details)),
+            None => {
+                if request.device_channel == DeviceChannel::Browser {
+                    Err(errors::ConnectorError::MissingRequiredField {
+                        field_name: "browser_info",
+                    })?
+                } else {
+                    Ok(None)
+                }
+            }
+        }?;
+        let card_details = get_card_details(request.payment_method_data.clone())?;
+        let currency = request
+            .currency
+            .map(|currency| currency.to_string())
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)
+            .into_report()
+            .attach_printable("missing field currency")?;
+        let purchase_currency: Currency = iso_currency::Currency::from_code(&currency)
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)
+            .into_report()
+            .attach_printable("error while parsing Currency")?;
+        let billing_address = request.billing_address.address.clone().ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "billing_address.address",
+            },
+        )?;
+        let billing_state = billing_address.clone().to_state_code()?;
+        let billing_country = isocountry::CountryCode::for_alpha2(
```

Comment:
why do we need this conversion?

Reply:
we need to get the numeric_id of the country (https://docs.rs/isocountry/latest/isocountry/numeric/index.html), that's the reason we are using isocountry